src/dodal/common/udc_directory_provider.py

```diff
@@ -20,7 +24,13 @@ def __init__(self, directory: Path | None = None):
             else None
         )
 
-    def update(self, directory: Path):
+    def update(self, directory: Path, create_directory=False):
+        if create_directory:
+            panda_directory = f"{directory}/{self.resource_dir}"
+            # At some point in the future, ophyd_async will have the feature to create the requested directory
+            if not os.path.isdir(panda_directory):
```

Comment:
This is in the Hyperion/blueapi container: which user does Hyperion run as? blueapi runs as a kubernetes user without file system permissions as far as possible

Reply:
Ah I hadn't considered that, Hyperion is running on a control machine so should have permissions. it might be better to just stick this part within a Hyperion plan

Reply:
Running as gda2?

Either way, if it's in the plan or the DirProvider it'll need to be running as a user that can write to the visit directory

Reply:
Yeah, as gda2. We can either check, in dodal, if the user has permission to create a directory if `create_directory=True` before trying to create it, or have the logic in Hyperion where the user will always be able to. I think Hyperion is better for now, can always change later

Reply:
Moved the logic to Hyperion so this PR is pretty tiny now!